libcxx/include/__format/format_arg_store.h

```diff
@@ -32,6 +33,12 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 
 namespace __format {
 
+template <class _Arr, class _Elem>
+inline constexpr bool __is_bounded_array_of = false;
+
+template <class _Elem, size_t _Len>
+inline constexpr bool __is_bounded_array_of<_Elem[_Len], _Elem> = true;
+
```

Comment:
Are these really useful? They are only used twice.

Reply:
I roughly remember that the old form was related to something weird (possibly for the `T[0]` extension), so I added the new internal trait.